airlock/views/request.py

```diff
@@ -323,12 +270,68 @@ def request_view(request, request_id: str, path: str = ""):
         "multiselect_withdraw": release_request.is_editing(),
         "code_url": code_url,
         "return_url": "",
-        "group_comment_delete_url": group_comment_delete_url,
     }
 
     return TemplateResponse(request, template, context)
 
 
+def group_presenter(release_request, relpath, request):
+    """Present to build group template context, which is needed in most request views."""
+
+    assert relpath != ROOT_PATH
+
+    group = relpath.parts[0]
+    filegroup = release_request.filegroups.get(group)
+    visibilities = release_request.get_writable_comment_visibilities_for_user(
+        request.user
+    )
+    # are we on the group page?
+    if len(relpath.parts) == 1:
+        inline = False
+    else:
+        inline = True
+
+    # are comments readonly?
+    c2_readonly = inline or not permissions.user_can_edit_request(
+        request.user, release_request
+    )
```

Comment:
looks like c2 is context & controls, not comments?

```suggestion
    # are context and controls readonly?
    c2_readonly = inline or not permissions.user_can_edit_request(
        request.user, release_request
    )
```

Reply:
(this is why I like extra-long variable names...)

Reply:
> Curious - you would seriously prefer context_and_controls_readonly here?

yes I would! I know I'm an outlier...